### PR TITLE
luajit-openresty: add version_scheme, livecheck

### PR DESCRIPTION
**Pull request by samford**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Homebrew/brew#11491 will add a `Version` parser that accounts for filenames which don't include a leading software name (e.g., `brew-1.2.3.tar.gz` vs. `v2.1-20210510.tar.gz` here). As described in that PR, the parsed version for `luajit-openresty` only includes the trailing date (`20210510`) instead of the full version (`2.1-20210510`). This occurs because the existing `Version` parsers assume that anything before the hyphen in a filename like `v2.1-20210510.tar.gz` is the leading software name and omits it. This isn't true for `luajit-openresty` and the brew PR accounts for this type of filename format.

The aforementioned brew PR will lead to the full version (`2.1-20210510`) being used for `luajit-openresty`, so this PR bumps `version_scheme`.

This also adds a `livecheck` block with a permissive regex that can match unstable versions (since the latest LuaJIT version [from 2017] is a beta). We can revisit this if the regex ends up being too loose over time (matching something we don't want) but I feel that it's better to be permissive in this particular instance, since the formula is technically using unstable versions.

---

For context on this, see #78791 and Homebrew/brew#11491, where we discussed the appropriateness and benefits/detriments of using only the date as the version or using the full version that OpenResty provides.

The arguments for only using the date as the version are: LuaJIT will never make another release (people have asked and it won't happen), the `2.1-` prefix is just noise (it will never change), and we would need to bump `version_scheme` for this to work (as seen in this PR).

My arguments for using the full version can be summarized as follows:

* Upstream uses a tag format like `v2.1-20210510` and it's not our place to say that the `v2.1-` prefix isn't a meaningful part of the version simply because it's unlikely to change. [Other package managers use the full version](https://repology.org/project/luajit-openresty/versions) but we're the lone exception and this is only because of a shortcoming in our `Version` parser. We should use the version format that upstream has provided and this is in keeping with how we handle this type of situation in other formulae.
* LuaJIT is an actively-developed project and even if they've said that they won't make a new release, I don't see any reason why we should encode this assumption into how we handle the `luajit-openresty` version (see previous point). It doesn't seem unreasonable to me that LuaJIT could reverse course and make a new release sometime in the future (people change their mind, the composition of the development team and/or leadership could change, etc.).
* Using the full version (`2.1-20210510`) will work both now and in the future for foreseeable potentialities (e.g., if LuaJIT ever released a `2.2.0` version). If we only use the date as the version, `20210510` would be erroneously treated as newer than a version like `2.2.0`. A date version only works if OpenResty uses the `2.1-20210510` format forever. This isn't a given, as the repository used tags like `v2.0.5` in the past and switched to the date-based format after `v2.1.0-beta3` because LuaJIT stopped making releases. It's not unreasonable to believe that OpenResty could return to a format like `v2.2.0` if LuaJIT made such a release.

Basically, using the full version that upstream provides is appropriate here but it would also adapt better to potential version changes in the future (however unlikely they are). We bump `version_scheme` once here and we should be good for the foreseeable future, unless something really unexpected happens upstream (which we would need to address regardless).

### CODE REVIEW

**Comment by samford** [comment]
I've cancelled the CI run because there's no point in running it until Homebrew/brew#11491 is merged.

**Comment by carlocab** [comment]
This doesn't affect the build, so this can be a syntax-only PR. Added the label.

**Comment by samford** [comment]
> This doesn't affect the build, so this can be a syntax-only PR.

My thinking was that something might have to happen in relation to bottles since the version will change. For example, running `brew install luajit-openresty` using the brew PR and this one, I get:

```
==> Downloading https://ghcr.io/v2/homebrew/core/luajit-openresty/manifests/2.1-
##O=#  #
curl: (22) The requested URL returned error: 404
Error: Failed to download resource "luajit-openresty_bottle_manifest"
Download failed: https://ghcr.io/v2/homebrew/core/luajit-openresty/manifests/2.1-20210510
```

My knowledge of the bottle situation on GitHub is admittedly lacking but is this something that would still be addressed by a syntax-only PR?

**Comment by carlocab** [comment]
No, you're right. `version_scheme` bumps are like `revision` bumps here. They affect the version of the built bottles. We should build bottles here.

**Comment by samford** [comment]
Homebrew/brew#11491 is now merged but we have to wait until it's part of a tagged `brew` version before this PR can be merged. I'll rebase this and let CI run after the next `brew` release.

**Comment by samford** [comment]
Homebrew/brew#11491 is now part of version `3.2.0`, which was released today. I've rebased this and removed the temporary `version` that was added to maintain the previous version scheme until this PR bumping `version_scheme` could be merged.

I believe this should finally be good to merge once it passes CI.

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**
